qiskit_ionq/ionq_job.py

```diff
@@ -410,39 +410,60 @@ def _format_result(self, data):
             if metadata.get("sampler_seed", "").isdigit()
             else None
         )
-        qiskit_header = decompress_metadata_string_to_dict(
-            metadata.get("qiskit_header", None)
+        qiskit_header = decompress_metadata_string(metadata.get("qiskit_header", None))
+        if not isinstance(qiskit_header, list):
+            qiskit_header = [qiskit_header]
+        shots = (
+            int(metadata.get("shots", 1024))
+            if str(metadata.get("shots", "1024")).isdigit()
+            else 1024
         )
-
-        shots = int(metadata.get("shots") if metadata.get("shots").isdigit() else 1024)
-        job_result = {
-            "data": {},
-            "shots": shots,
-            "header": qiskit_header or {},
-            "success": success,
-        }
-        if self._status == jobstatus.JobStatus.DONE:
-            (counts, probabilities) = _build_counts(
-                data,
-                self._num_qubits,
-                self._clbits,
-                shots,
-                use_sampler=is_ideal_simulator,
-                sampler_seed=sampler_seed,
-            )
-            job_result["data"] = {
-                "counts": counts,
-                "probabilities": probabilities,
-                # Qiskit/experiments relies on this being present in this location in the
-                # ExperimentData class.
-                "metadata": qiskit_header or {},
+        job_result = [
+            {
+                "data": {},
+                "shots": shots,
+                "header": qiskit_header[i] or {},
```

Comment:
I'm not certain the ordering of results and metadata is consistent

Reply:
Maybe serialize as circuit-hash like you first thought?

Reply:
ill try using the circuit name first hopefully that works